problem_855_a/Problem.java

```diff
@@ -0,0 +1,25 @@
+import java.util.Scanner;
+import java.util.HashSet;
+
+public class Problem {
+  public static void main(String[] a) {
+    Scanner sc = new Scanner(System.in);
+    int possessionTimes = sc.nextInt();
+    sc.nextLine();
+    HashSet<String> names = new HashSet<>();
+    String output = "";
+
+    for (int i = 0; i < possessionTimes; i++) {
+      String input = sc.nextLine();
+      if (!names.contains(input)) {
+        names.add(input);
+        output += "NO\n";
```

Comment:
Don't do like this in production code. It is performance issue. Every concatenation creates new String in the String pool, because strings are immutable. For such kind of tasks, always try to use StringBuilder.

If you don't know what is String Pool and so on, you must read and learn about it.